diff --git a/src/lib/web.ts b/src/lib/web.ts
index fc1a782f1c2a677d001e3ba89913b438564244b3..4d307ca72649cfead165cecf1ad979a3e6bbfe4e 100644
--- a/src/lib/web.ts
+++ b/src/lib/web.ts
@@ -59,7 +59,7 @@ export async function toInternalRequest(
       action,
       providerId,
       method: req.method,
-      headers: Object.fromEntries(req.headers),
+      headers: Object.fromEntries(req.headers.entries()),
       body: req.body ? await getBody(req) : undefined,
       cookies: parseCookie(req.headers.get("cookie") ?? "") ?? {},
       error: url.searchParams.get("error") ?? undefined,